framework/src/outputs/Checkpoint.C

```diff
@@ -102,10 +113,23 @@ Checkpoint::outputStep(const ExecFlagType & type)
 bool
 Checkpoint::shouldOutput(const ExecFlagType & type)
 {
+
   // Check if the checkpoint should "normally" output, i.e. if it was created
   // through checkpoint=true
   bool shouldOutput = (onInterval() || type == EXEC_FINAL) ? FileOutput::shouldOutput(type) : false;
 
+  // Check if we should output an autosave based on a time interval.
+  auto curr_time = std::chrono::steady_clock::now();
+  auto elapsed = std::chrono::duration_cast<std::chrono::seconds>(curr_time - start_time);
+  // Print only on timestep end to avoid weird issues
+  if (elapsed >= _autosave_time_interval and type == EXEC_TIMESTEP_END)
+  {
+    start_time = std::chrono::steady_clock::now();
+    _console << std::to_string(_autosave_time_interval.count()) +
```

Comment:
Using "to_string" is ok, but I'll point out that streaming operators and all of our macros handle mixed types natively so it's unecessary. You can simply say put `(_autosave...count() << " seconds have elapsed...")`